src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -32,6 +32,8 @@ public class TagHelperDescriptorFactory
                 RegexOptions.None,
                 Constants.RegexMatchTimeout);
 
+        private static readonly char[] UnsupportedAttributeSelectors = new[] { '~', '|', '*' };
```

Comment:
> everything prior to the `=` not captured by the blacklist

I think you're correct and the correct whitelist is enforced elsewhere. So remove the unnecessary code using this blacklist.